src/java/org/apache/cassandra/service/reads/AbstractReadExecutor.java

```diff
@@ -155,7 +156,11 @@ private void makeRequests(ReadCommand readCommand, Iterable<Replica> replicas)
         if (hasLocalEndpoint)
         {
             logger.trace("reading {} locally", readCommand.isDigestQuery() ? "digest" : "data");
-            Stage.READ.maybeExecuteImmediately(new LocalReadRunnable(command, handler));
+
+            if (TEST_FORCE_ASYNC_LOCAL_READS)
+                new Thread(new LocalReadRunnable(readCommand, handler)).start();
```

Comment:
Initially I tried to just have this hit `Stage.READ.execute()`, but the initial read and the read on read-repair seemed to both try to use the same ReadStage thread :/

Reply:
That sounds better than what I've got here. I'll give it a shot...